File: src/code/kotlin/LengthOfLastWord.kt
package code.kotlin

class LengthOfLastWord {
    fun lengthOfLastWord(s: String): Int {
        if (s.length <= 1)
            return s.length

        var i = s.length - 1

        while (i >= 0 && s[i] == ' ')
            i--

        val start = i

        while (i >= 0 && s[i] != ' ')
            i--

        return start - i
    }
}

fun main() {
    val solution = LengthOfLastWord()
    println(solution.lengthOfLastWord("   fly me   to   the moon  "))
}

File: src/code/python/length_of_last_word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i = len(s) - 1

        while i >= 0 and s[i] == ' ':
            i -= 1

        start = i

        while i >= 0 and s[i] != ' ':
            i -= 1

        return start - i


if __name__ == '__main__':
    sol = Solution()
    print(sol.lengthOfLastWord("Hello World"))
    print(sol.lengthOfLastWord("   fly me   to   the moon  "))
    print(sol.lengthOfLastWord("luffy is still joyboy"))
